packages/medusa/src/types/pricing.ts

```diff
@@ -61,7 +60,7 @@ export type ShippingOptionPricing = {
  *         type: number
  *         description: The taxes applied.
  */
-export type PricedShippingOption = Partial<ShippingOption> &
+export type PricedShippingOption = Omit<ShippingOption, "beforeInsert"> &
```

Comment:
Note: instead of make all fields optional, only omit the private method which prevented from efficiently using PricedShippingOption